development/build/display.js

```diff
@@ -0,0 +1,156 @@
+const randomColor = require('randomcolor')
+const chalk = require('chalk')
+
+module.exports = { setupTaskDisplay, displayChart }
+
+const SYMBOLS = {
+  Empty: '',
+  LeftCorner: '◺',
+  RightCorner: '◿',
+  Space: ' ',
+  Full: '█',
+  SevenEighths: '▉',
+  ThreeQuarters: '▊',
+  FiveEighths: '▋',
+  Half: '▌',
+  ThreeEighths: '▍',
+  Quarter: '▎',
+  Eighth: '▏',
+  EOL: require('os').EOL,
+  RightHalf: '▐',
+  RightEigth: '▕',
+}
+
+function setupTaskDisplay (taskEvents) {
+  const taskData = []
+  taskEvents.on('start', ([name]) => {
+    console.log(`Starting '${name}'...`)
+  })
+  taskEvents.on('end', ([name, start, end]) => {
+    taskData.push([name, start, end])
+    console.log(`Finished '${name}'`)
+  })
+  taskEvents.on('complete', () => {
+    displayChart(taskData)
+  })
+}
+
+function displayChart (data) {
+  // sort tasks by start time
+  data.sort((a, b,) => a[1] - b[1])
+
+  // get bounds
+  const first = Math.min(...data.map((entry) => entry[1]))
+  const last = Math.max(...data.map((entry) => entry[2]))
+
+  // get colors
+  const colors = randomColor({ count: data.length })
+
+  // some heading before the bars
+  console.log(`\nbuild completed. task timeline:`)
+
+  // build bars for bounds
+  data.map((entry, index) => {
+    const [label, start, end] = entry
+    const [start2, end2] = [start, end].map((value) => adjust(value, first, last, 40))
+    const barString = barBuilder(start2, end2)
+    const color = colors[index]
+    const coloredBarString = colorize(color, barString)
+    const duration = ((end - start) / 1e3).toFixed(1)
+    console.log(coloredBarString, `${label} ${duration}s`)
+  })
+
+}
+
+function colorize (color, string) {
+  const colorizer = (typeof chalk[color] === 'function') ? chalk[color] : chalk.hex(color)
+  return colorizer(string)
+}
+
+// scale number within bounds
+function adjust (value, first, last, size) {
+  const length = last - first
+  const result = (value - first) / length * size
+  return result
+}
+
+// draw bars
+function barBuilder (start, end) {
+  const [spaceInt, spaceRest] = splitNumber(start)
+  const barBodyLength = end - spaceInt
+  let [barInt, barRest] = splitNumber(barBodyLength)
+  // console.log(`${barBodyLength} ${barInt}`)
+  // We are handling zero value as a special case
+  // to print at least something on the screen
+  if (barInt === 0 && barRest === 0) {
+    barInt = 0
+    barRest = 0.001
+  }
+
+  const spaceFull = SYMBOLS.Space.repeat(spaceInt)
+  const spacePartial = getSymbolNormalRight(spaceRest)
+  const barFull = SYMBOLS.Full.repeat(barInt)
+  const barPartial = getSymbolNormal(barRest)
+  // console.log(`${spaceFull.length} ${spacePartial.length} ${barFull.length} ${barPartial.length}`)
+
+  return `${spaceFull}${spacePartial}${barFull}${barPartial}`
```

Comment:
There seems to be a rounding error somewhere 🤔 Locally it looks like `zip` starts consistently _before_ the JS bundles finish, even though they finish at the exact same millisecond that zip starts.

Reply:
Oh I see - it's because there are fewer right-aligned block characters. That's a shame.

Reply:
i realized the correct way to do this after is to mix left-aligned block chars with inverted text/background colors